### PR TITLE
fix: respect processor group configuration for copybookName autocompl…

### PR DESCRIPTION
**Pull request by amanPras**
…etion

## How Has This Been Tested?

- [x] Test copybook Name completion for a dialect with configuration present in only processor groups (No settings.json)

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by ishche — APPROVED** [review]
Maybe at some point in the future we should remove fetchTextConfiguration method at all.

File: server/engine/src/test/java/org/eclipse/lsp/cobol/service/copybooks/CopybooksResolutionWithVariableInPathTest.java [hunk 1]
```diff
@@ -15,48 +15,50 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.mockito.ArgumentMatchers.*;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-/**
- * Tests for copybook name resolution with variables in paths.
- */
+/** Tests for copybook name resolution with variables in paths. */
 class CopybooksResolutionWithVariableInPathTest {
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/service/copybooks/CopybooksResolutionWithVariableInPathTest.java:37]: @ishche Can we test this code on Mac? This test is used to break on Windows machines.

**Review by Nurkambay — APPROVED** [review]

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/copybooks/CopybookNameServiceImpl.java [hunk 2]
```diff
@@ -86,25 +85,13 @@ public Optional<CopybookName> findByName(String uri, final String displayName) {
...
     copybookLocalFolders.add(
-            settingsService.fetchConfigurations(documentUri, ImmutableList.of(CPY_LOCAL_PATHS.label))
-                    .thenApply(unwrapJsonArrays));
-    return settingsService.fetchTextConfiguration(DIALECTS.label)
+            settingsService.fetchTextConfigurationWithScope(documentUri, CPY_LOCAL_PATHS.label));
+    return settingsService.fetchTextConfigurationWithScope(documentUri, DIALECTS.label)
             .thenAccept(
                     dialects -> dialects.forEach(dialect -> copybookLocalFolders.add(
-                            settingsService.fetchTextConfiguration(
+                            settingsService.fetchTextConfigurationWithScope(documentUri,
```
Comment (amanPras) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/service/copybooks/CopybookNameServiceImpl.java]: this is not supported by processor group settings

File: clients/cobol-lsp-vscode-extension/src/constants.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ export const SETTINGS_TAB_CONFIG: string = "cobol-lsp.smart-tab";
 
 export const SERVER_PORT = "cobol-lsp.server.port";
 export const SERVER_RUNTIME = "cobol-lsp.serverRuntime";
+export const DIALECT_LIBS = "cobol-lsp.dialect.libs";
```
Comment (amanPras) [inline clients/cobol-lsp-vscode-extension/src/constants.ts:30]: This is a new request that the server asks the client. This request also takes dialect as a parameter. The client handles this request in a similar manner as for copybooks for consistency across the project. It looks into the settings.json as well as proc_grps.

**Review by ishche — APPROVED** [review]

**Pull request merged**